### ISSUE TITLE
Feature request

### ISSUE THREAD

**Issue by DRomatzki** [paul-gauthier/aider#1030]
### Issue

For us poor guys who run Ollama...... we are mostly limited to the smaller models, the 7b and 8b models.
I have tried to develop an App and Aider bombed out after the limit of iterations has been reached and could not get further.
A possible solution would be to have a list of available models and then, when the limit is reached, the model can be automatically changed and the possible loop resolved.
This can even be of value to online service providers because non is 100% perfect and also gets into loops, That is my experience also with Claude, ChatGPT and Deepcoder. I sometimes have to switch between them to resolve a loop.
If you have the time can you please implement such a feature?

**Comment by paul-gauthier** [comment]
Thanks for trying aider and filing this issue.

Unfortunately, the local quantized ollama 7-8b models are probably not capable enough to follow aider's system prompt.

**Comment by DRomatzki** [comment]
Yes and no. 
First off .....
I have been using the following for free: claude. chatgpt, deepseek coder V2 and even perplexity. I have experienced loops in all of them, What I then do is just copy the code from service provider/model to the next and continue. Claude was the best, then deepseekcode and chatgpt where midlevel and perplexity is not usefull but for simple questions. But all had loops and could not complete some code. That was using for nodejs and typescrypt code.
I did try to use aider with Ollama with deepseekcoder V2 and llama3 with limited success. However, I believe there are models that can be fine tuned for coding and reach better results, even at 7B or 8B. I am going to try the finetuning option if I can find a dataset to use for, say, javascrypt or something.
Even if you consider the 7B and 8B models insufficient, such a feature will benefit others using the online bigger models.
But that is just my 2 cents .....
But I must say.... Aider is great and you did a great job......
